### ISSUE TITLE
after voting, the expected producers didn't produce

### ISSUE THREAD

**Issue by yankunsam** [EOSIO/eos#5346]
In my private network. I set the bios/token/system contract. I create some accounts and delegatebw, vote. But the schedule did work.

```
sam@wuyou-0:~/Public/Eoasis/oasis$ cleos get table eosio eosio voters
{
  "rows": [{
      "owner": "samkunnbanb3",
      "proxy": "",
      "producers": [],
      "staked": 40000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    },{
      "owner": "samkunnbanb4",
      "proxy": "",
      "producers": [
        "samkunnbanb3"
      ],
      "staked": "2520000040000",
      "last_vote_weight": "1067445845383733632.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    }
  ],
  "more": false
}
```

```
sam@wuyou-0:~/Public/Eoasis/oasis$ cleos get table eosio eosio producers
{
  "rows": [{
      "owner": "samkunnbanb1",
      "total_votes": "0.00000000000000000",
      "producer_key": "EOS5qwCiynJ9ip37aE2w3md9pWKCTk59wv2MffhfKk7KQvcd8c3bC",
      "is_active": 1,
      "url": "www.samkunnbanb1",
      "unpaid_blocks": 0,
      "last_claim_time": 0,
      "location": 123
    },{
      "owner": "samkunnbanb2",
      "total_votes": "0.00000000000000000",
      "producer_key": "EOS7pQ4ubV6R4mYkbSDpuq5nw2KPufQdxshREDYwk1LEoKS3NMNDe",
      "is_active": 1,
      "url": "www.samkunnbanb2",
      "unpaid_blocks": 0,
      "last_claim_time": 0,
      "location": 123
    },{
      "owner": "samkunnbanb3",
      "total_votes": "1067445845383733632.00000000000000000",
      "producer_key": "EOS6MkvSUBo7cVpmCKt2oqJKiqCuivizzGvyoKovB4eiGYs6zTUwB",
      "is_active": 1,
      "url": "www.samkunnbanb3",
      "unpaid_blocks": 0,
      "last_claim_time": 0,
      "location": 123
    }
  ],
  "more": false
}
sam@wuyou-0:~/Public/Eoasis/oasis$
```

```
sam@wuyou-0:~/Public/Eoasis/oasis$ cleos get currency stats eosio.token EOS
{
  "EOS": {
    "supply": "1000000000.0000 EOS",
    "max_supply": "1000000000.0000 EOS",
    "issuer": "eosio.token"
  }
}
sam@wuyou-0:~/Public/Eoasis/oasis$
```

BTW,  after deploy bios contract, I push the 'setprods' action.

So is there some missing steps?   Thanks.

**Comment by jgiszczak** [comment]
The accounts you wish to produce must be named as `producer-name =` in the node config.ini, or passed on the command line as `--producer-name=` or `-p`.  The [multi-node](https://developers.eos.io/eosio-nodeos/docs/local-multi-node-testnet) guide describes this for the `inita` account.  You'll need the same argument(s), but with your account names instead of `inita`.  A single node can support all three of your accounts as producers if you wish.  Just repeat the `producer-name` argument/config, one for each name.

If you have multiple synched nodes, you may distribute the `producer-name` configuration any way you like.  All on one node, one on each node, or some on one node and some on another.  The only thing you may not do is repeat the same name on multiple nodes.

**Comment by yankunsam** [comment]
@jgiszczak  Thanks for your reply. I do this in the config.ini. It works after eosio.bios  contract when I set the specific producers in setprods action. But after deploying the eosio.system contract and the delegatebw and voting over 15%, the elected producers did produce and there is no changes after 1min. BTW, I set the private network in 4 physical machines.

**Comment by yankunsam** [comment]
https://developers.eos.io/eosio-nodeos/docs/local-multi-node-testnet. It only focus multi node, not contains the voting process, like main net booting.

**Comment by yankunsam** [comment]
```

# ID of producer controlled by this node (e.g. inita; may specify multiple times) (eosio::producer_plugin)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1,1           Top
# the endpoint upon which to listen for incoming connections (eosio::bnet_plugin)
bnet-endpoint = 0.0.0.0:4321

# the number of threads to use to process network messages (eosio::bnet_plugin)
# bnet-threads =

# remote endpoint of other node to connect to; Use multiple bnet-connect options as needed to compose a network (eosio::bnet_plugin)
# bnet-connect =

# this peer will request no pending transactions from other nodes (eosio::bnet_plugin)
bnet-no-trx = false

# the location of the blocks directory (absolute path or relative to application data dir) (eosio::chain_plugin)
blocks-dir = "blocks"

# Pairs of [BLOCK_NUM,BLOCK_ID] that should be enforced as checkpoints. (eosio::chain_plugin)
# checkpoint =

# Override default WASM runtime (eosio::chain_plugin)
# wasm-runtime =

# Maximum size (in MB) of the chain state database (eosio::chain_plugin)
chain-state-db-size-mb = 1024

# Maximum size (in MB) of the reversible blocks database (eosio::chain_plugin)
reversible-blocks-db-size-mb = 340

# print contract's output to console (eosio::chain_plugin)
contracts-console = true

# Account added to actor whitelist (may specify multiple times) (eosio::chain_plugin)
# actor-whitelist =

# Account added to actor blacklist (may specify multiple times) (eosio::chain_plugin)
# actor-blacklist =

# Contract account added to contract whitelist (may specify multiple times) (eosio::chain_plugin)
# contract-whitelist =

# Contract account added to contract blacklist (may specify multiple times) (eosio::chain_plugin)
# contract-blacklist =

# Track actions which match receiver:action:actor. Actor may be blank to include all. Receiver and Action may not be blank. (eosio::history_plugin)
#filter-on = :transfer:

# PEM encoded trusted root certificate (or path to file containing one) used to validate any TLS connections made.  (may specify multiple times)
#  (eosio::http_client_plugin)
# https-client-root-cert =

# true: validate that the peer certificates are valid and trusted, false: ignore cert errors (eosio::http_client_plugin)
https-client-validate-peers = 1

# The local IP and port to listen for incoming http connections; set blank to disable. (eosio::http_plugin)
http-server-address = 0.0.0.0:8888

# The local IP and port to listen for incoming https connections; leave blank to disable. (eosio::http_plugin)
# https-server-address =

# Filename with the certificate chain to present on https connections. PEM format. Required for https. (eosio::http_plugin)
# https-certificate-chain-file =

# Filename with https private key in PEM format. Required for https (eosio::http_plugin)
# https-private-key-file =

# Specify the Access-Control-Allow-Origin to be returned on each request. (eosio::http_plugin)
access-control-allow-origin = *

# Specify the Access-Control-Allow-Headers to be returned on each request. (eosio::http_plugin)
access-control-allow-headers = *

# Specify the Access-Control-Max-Age to be returned on each request. (eosio::http_plugin)
# access-control-max-age =

# Specify if Access-Control-Allow-Credentials: true should be returned on each request. (eosio::http_plugin)
access-control-allow-credentials = false

# The actual host:port used to listen for incoming p2p connections. (eosio::net_plugin)
p2p-listen-endpoint = 0.0.0.0:9876

# An externally accessible host:port for identifying this node. Defaults to p2p-listen-endpoint. (eosio::net_plugin)
# p2p-server-address =

# The public endpoint of a peer node to connect to. Use multiple p2p-peer-address options as needed to compose a network. (eosio::net_plugin)
p2p-peer-address = 192.168.1.186:9876
p2p-peer-address = 192.168.1.183:9876
p2p-peer-address = 192.168.1.243:9876


# Maximum number of client0nodes from any single IP address (eosio::net_plugin)
p2p-max-nodes-per-host = 1

# The name supplied to identify this node amongst the peers. (eosio::net_plugin)
agent-name = "EOS Test Agent"

# Can be 'any' or 'producers' or 'specified' or 'none'. If 'specified', peer-key must be specified at least once. If only 'producers', peer-key is not required. 'producers' and 'specified' may be combined. (eosio::net_plugin)
allowed-connection = any

# Optional public key of peer allowed to connect.  May be used multiple times. (eosio::net_plugin)
# peer-key =

# Tuple of [PublicKey, WIF private key] (may specify multiple times) (eosio::net_plugin)
# peer-private-key =

# Maximum number of clients from which connections are accepted, use 0 for no limit (eosio::net_plugin)
max-clients = 25

# number of seconds to wait before cleaning up dead connections (eosio::net_plugin)
connection-cleanup-period = 30

# True to require exact match of peer network version. (eosio::net_plugin)
network-version-match = 0

# number of blocks to retrieve in a chunk from any individual peer during synchronization (eosio::net_plugin)
sync-fetch-span = 100

# maximum sizes of transaction or block messages that are sent without first sending a notice (eosio::net_plugin)
max-implicit-request = 1500

# Enable block production, even if the chain is stale. (eosio::producer_plugin)
enable-stale-production = true

# Start this node in a state where production is paused (eosio::producer_plugin)
pause-on-startup = false

# Limits the maximum time (in milliseconds) that is allowed a pushed transaction's code to execute before being considered invalid (eosio::producer_plugin)
max-transaction-time = 30

# Limits the maximum age (in seconds) of the DPOS Irreversible Block for a chain this node will produce blocks on (use negative value to indicate unlimited) (eosio::producer_plugin)
max-irreversible-block-age = -1

# ID of producer controlled by this node (e.g. inita; may specify multiple times) (eosio::producer_plugin)
producer-name = samkunnbanb3
#producer-name = eosio

# (DEPRECATED - Use signature-provider instead) Tuple of [public key, WIF private key] (may specify multiple times) (eosio::producer_plugin)
# private-key =

# Key=Value pairs in the form <public-key>=<provider-spec>
# Where:
#    <public-key>       is a string form of a vaild EOSIO public key
#
#    <provider-spec>    is a string in the form <provider-type>:<data>
#
#    <provider-type>    is KEY, or KEOSD
#
#    KEY:<data>         is a string form of a valid EOSIO private key which maps to the provided public key
#
#    KEOSD:<data>       is the URL where keosd is available and the approptiate wallet(s) are unlocked (eosio::producer_plugin)
signature-provider = EOS6MkvSUBo7cVpmCKt2oqJKiqCuivizzGvyoKovB4eiGYs6zTUwB=KEY:5KJ6ygSpfjfTqQ58bzqwGTJN7wA8y5Rgy8WFkZDub4ZxcBf7ZAW
#signature-provider = EOS56epkkpVdmZ49UhSKJagh5F6qgBeEFkHyL25Z2Z1a4SeUqRbm7=KEY:5Hz2G2L9p3k7YhkqGJaioNJQnYKjtZKKS2y3wRLDobCQAtXg5oA

# Limits the maximum time (in milliseconds) that is allowd for sending blocks to a keosd provider for signing (eosio::producer_plugin)
keosd-provider-timeout = 5

# Lag in number of blocks from the head block when selecting the reference block for transactions (-1 means Last Irreversible Block) (eosio::txn_test_gen_plugin)
txn-reference-block-lag = 0

# The path of the wallet files (absolute path or relative to application data dir) (eosio::wallet_plugin)
wallet-dir = "/home/sam/eosio-wallet/sam.wallet"

# Timeout for unlocked wallet in seconds (default 900 (15 minutes)). Wallets will automatically lock after specified number of seconds of inactivity. Activity is defined as any wallet command e.g. list-wallets. (eosio::wallet_plugin)
unlock-timeout = 9000

# eosio key that will be imported automatically when a wallet is created. (eosio::wallet_plugin)
# eosio-key =

# Plugin(s) to enable, may be specified multiple times
plugin = eosio::chain_plugin
plugin = eosio::chain_api_plugin
plugin = eosio::http_plugin
plugin = eosio::chain_api_plugin
plugin = eosio::producer_plugin
```

**Comment by taokayan** [comment]
have you passed the 15% activation threshold, which is 150,000,000 EOS must be staked and voted.

**Comment by yankunsam** [comment]
Of course. I create two account and vote for others 

```
sam@wuyou-0:~/Public/Eoasis/oasis$ cleos get account samkunnbanb5
permissions:
     owner     1:    1 EOS5ti193hu1ntsHbMRBC8Z1rCUZsUZMbgpMQ3Fu2TzBEq8vHzzk7
        active     1:    1 EOS5ti193hu1ntsHbMRBC8Z1rCUZsUZMbgpMQ3Fu2TzBEq8vHzzk7
memory:
     quota:     133.5 KiB    used:     3.514 KiB

net bandwidth:
     staked:   50000001.0000 EOS           (total stake delegated from account to self)
     delegated:       0.0000 EOS           (total staked delegated to account from others)
     used:               216 bytes
     available:        488.2 GiB
     limit:            488.2 GiB

cpu bandwidth:
     staked:   50000001.0000 EOS           (total stake delegated from account to self)
     delegated:       0.0000 EOS           (total staked delegated to account from others)
     used:             6.216 ms
     available:        27.73 hr
     limit:            27.73 hr

EOS balances:
     liquid:    100000000.0000 EOS
     staked:    100000002.0000 EOS
     unstaking:         0.0000 EOS
     total:     200000002.0000 EOS

producers:
     samkunnbanb1    samkunnbanb3    samkunnbanb4

sam@wuyou-0:~/Public/Eoasis/oasis$ cleos get account samkunnbanb4
permissions:
     owner     1:    1 EOS7WY8N3f5Gv5ioRBpHYSkAFhFKzvwZCynvVewXH9aRbnCzLWhkT
        active     1:    1 EOS7WY8N3f5Gv5ioRBpHYSkAFhFKzvwZCynvVewXH9aRbnCzLWhkT
memory:
     quota:     133.5 KiB    used:     3.934 KiB

net bandwidth:
     staked:   30000001.0000 EOS           (total stake delegated from account to self)
     delegated:       0.0000 EOS           (total staked delegated to account from others)
     used:               368 bytes
     available:        294.1 GiB
     limit:            294.1 GiB

cpu bandwidth:
     staked:   30000001.0000 EOS           (total stake delegated from account to self)
     delegated:       0.0000 EOS           (total staked delegated to account from others)
     used:             7.578 ms
     available:        16.71 hr
     limit:            16.71 hr

EOS balances:
     liquid:     40000000.0000 EOS
     staked:     60000002.0000 EOS
     unstaking:         0.0000 EOS
     total:     100000002.0000 EOS

producers:
     samkunnbanb1    samkunnbanb3    samkunnbanb4

sam@wuyou-0:~/Public/Eoasis/oasis$
```

**Comment by yankunsam** [comment]
It works now. Thanks. Maybe the first time, I use only one account to voting. And this is not stake activated.
